include/inviwo/core/util/transformiterator.h

```diff
@@ -91,6 +94,13 @@ struct TransformIterator {
         return *this;
     }
 
+    template <typename Other>
+    TransformIterator(const Other& other)
+        requires(!std::is_same_v<TransformIterator, Other>) &&
```

Comment:
Why must Other be different?

Reply:
if they are the same the copy constructor will be used.